### PR TITLE
Improve Windows default editor detection

### PR DESCRIPTION
**Pull request by brandon93s**
The current default editor check for Windows suffers from false-positives. The false-positives result in the config being opened with WScript and the user being prompted with an error:

image

The current implementation checks _all_ values in `...FileExts\\.js`. The updates in this PR filter down the values to just those inside of `...FileExts\\.js\\UserChoice` which is where the user's configured editor lives. `OpenWithList`, which powers the right click -> open with context menu, is one of the sub-keys. This can be populated with valid targets without the user having ever provided a default program:

image

The shell does not use values from this list. By including them in the check, the code can wrongfully assume a default editor has been provided and proceeded without the notepad fallback. 

**TL;DR** - The check is more restrictive now, and should more accurately fallback to notepad avoiding errors. 

---

Tested:
- Unconfigured (opens notepad)
- WScript explicitly set as default (opens notepad)
- VS Code as default (opens code)
- Atom as default (opens atom)

### CODE REVIEW

**Comment by chabou** [comment]
Wow ! Awesome work @brandon93s! And very well documented PR 😍 

Thank you so much for your help! 🙏

**Comment by brandon93s** [comment]
Awesome! Thank you both for the quick review. 👍
